### DISCUSSION TITLE
Feature request: Support global files require

### DISCUSSION THREAD

**Discussion by nbabanov** [denoland/deno#13559] [Q&A]
**Example use case:**
I have to write tests for a deno app. I don't want to use the Deno API directly. Instead I want to use a custom Mocha pollyfill, which itself uses the Deno API. The file declares `globalThis.it = ...`, `globalThis.describe = ....` etc.

I want to be able to tell Deno to require this file and to provide the global variables/functions to every test file I have, so that I don't manually write import statements in all of my files.

**How to support it?**
What I envision as a support is to use the `files` key of the `tsconfig` which is supplied to Deno or some special Deno only key.

**Motivation**
This could save a lot of manual work in certain use cases.

**Comment by kitsonk** [comment]
You would create a module which places the runtime code in the global scope appropriately.  If you wanted the types to be available for runtime checking, you would need to declare them like:

```ts
declare global {
  var it: () => void;
  var describe: () => void;
}
```

You would then import it somewhere:

```ts
import "./myCustomPolyfill.ts";
```

Which will run the global code as well as declare the global types.

**Reply by nbabanov** [reply]
I don't want to have `import 'mocha-polyfill.ts'` in all of my files. I want to have a way to skip this overhead. As you know in the ususal mocha setup you are in mocha environment and you have special injected globals. I need a way to do just that. 

I have already commented this with @lucacasonato in Discord. There is no way currently for Deno to do that. Could you please convert this back to an `issue` and let us see how many people support this feature request?

**Reply by nbabanov** [reply]
Here is the pollyfill from Luca: https://gist.github.com/lucacasonato/54c03bb267074aaa9b32415dbfb25522#file-deno_mocha-ts

**Reply by zandaqo** [reply]
> I don't want to have import 'mocha-polyfill.ts' in all of my files.

It's been my understanding that test files are run in parallel--they are not in the same global environment. In that case, your only other option is to have a "master" file that imports all other test files along with the polyfill, that is, if you are willing to sacrifice parallelism for it.

**Reply by nbabanov** [reply]
I have also tought about this, but this has the same overhead if not bigger, because you need to maintain this entry file and as you said no parallelism.

**Reply by zandaqo** [reply]
A popular practice with test dependencies is to import all of them first into a single file ([dev_deps.ts](https://github.com/zandaqo/compago/blob/master/dev_deps.ts), for example) and then to import things as needed from the file in tests. We are usually importing assertions from STD anyway, so it doesn't make much difference. You can augment your global scope in the said dev_deps.ts as well.

**Reply by nbabanov** [reply]
Okay, this might do it.
